.github/workflows/release.yaml

```diff
@@ -94,3 +94,22 @@ jobs:
         if: env.SHOULD_DEPLOY_DOCS == 'true'
         id: deployment
         uses: actions/deploy-pages@v1
+
+      - name: Checkout API Docs
+        if: env.SHOULD_DEPLOY_DOCS == 'true'
+        uses: actions/checkout@v3
+        with:
+          ref: docs
+        
+      - name: Generate and Push API Docs
+        if: env.SHOULD_DEPLOY_DOCS == 'true'
+        run: |
+          pnpm install
```

Comment:
Shouldn't we ensure the `docs` branch is up-to-date with `master` before `install`?

Reply:
I couldn't understand how this was resolved, so I am re-opening this to ask:
 - How are you getting the new commits from `master` into `docs` before building it?

Reply:
Shouldn't it be a `merge` instead? Or, more specifically:

```
git merge origin master --no-edit
```

The `--no-edit` flag is to auto-accept the default commit message in a non-interactive manner.

Reply:
Agreed with `--no-edit` and will implement. I was doing a `pull` to ensure all `master` changes had been fetched.

Reply:
Yes, but the `pull` would `pull` the _remote_ `master` changes into the _local_ `master` branch, right? 😅 

The `merge` is to _remote_ `master` -> _local_ `docs`.

Reply:
To be fair, I am also happy using `merge` if that is the preferred implementation, I just use `pull` for this reason where base branch is the branch I am pulling, and do not need to investigate the fetched changes.